### PR TITLE
✨ feat: popover Component Token

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      popover Component Token     |
| 🇨🇳 Chinese |   popover Component Token        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e6e41f7</samp>

This pull request adds a component token demo and a migration guide for the popover component. It also updates the style file of the popover component to use component tokens instead of less variables. The component token feature is a new way to customize the appearance of the components using theme variables.

### <samp>🤖 Generated by Copilot at e6e41f7</samp>

*  Add a component token demo for the popover component to show how to customize its appearance using theme variables ([link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-4b8ee7a9586390091b32f1143f5d6b99c2b6d797a1e546f0022d9829364e7f34R1-R7), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-092798382a4a8a891d32ef1d207ac143aa01577715c58e41f70292203bfc9517R1-R41), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-be0bbc246cbe091c40db4aae74f6c5bae7920d84a1ff0024c68d6e3851e5d0cdR30), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-4adc3a014c480e4b095ef419f25dd05df6a317c5396cf8b83b98484c854246d6R31))
*  Modify the style file of the popover component to align the component token variables with the less variables that were previously used ([link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-f9107e2ff98d96f05f49b74223913c7382b375478e8f909fcd18679dc966302eL8-R21), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-f9107e2ff98d96f05f49b74223913c7382b375478e8f909fcd18679dc966302eL168-R180), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-f9107e2ff98d96f05f49b74223913c7382b375478e8f909fcd18679dc966302eL183-R199))
*  Add a mapping table of the less variables and the component token variables for the popover component in the documentation of how to migrate from less variables to component token variables ([link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL116-R131), [link](https://github.com/ant-design/ant-design/pull/42337/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L116-R131))

### CODE REVIEW

File: docs/react/migrate-less-variables.en-US.md [hunk 1]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@result-icon-font-size` | `iconFontSize` | - |
+| `@popover-bg` | `popoverBg` | - |
+| `@popover-color` | `popoverColor` | - |
+| `@popover-min-width` | `minWidth` | - |
+| `@popover-min-height` | `minHeight` | - |
+| `@popover-arrow-width` | `arrowWidth` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:126]: 这个用 `sizePopupArrow` 这个 global token 就好

File: docs/react/migrate-less-variables.en-US.md [hunk 2]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Pagination -->
 
-<!-- ### Popover -->
+### Popover>
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@result-icon-font-size` | `iconFontSize` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:121]: 这个可以不用带进来

File: docs/react/migrate-less-variables.en-US.md [hunk 3]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-bg` | `popoverBg` | - |
+| `@popover-color` | `popoverColor` | - |
+| `@popover-min-width` | `minWidth` | - |
+| `@popover-min-height` | `minHeight` | - |
+| `@popover-arrow-width` | `arrowWidth` | - |
+| `@popover-arrow-color` | `arrowColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:127]: 这个应该不需要了，现在 arrow 是跟着 popover 的 bg 一起变的，可以试一下

File: docs/react/migrate-less-variables.en-US.md [hunk 4]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-arrow-outer-color` | `arrowOuterColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:128]: 同上

File: docs/react/migrate-less-variables.en-US.md [hunk 5]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-distance` | `distance` | - |
+| `@popover-padding-horizontal` | `paddingHorizontal` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:130]: 这个废弃吧

File: docs/react/migrate-less-variables.en-US.md [hunk 6]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
 <!-- ### Pagination -->
 
-<!-- ### Popover -->
+### Popover>
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@result-icon-font-size` | `iconFontSize` | - |
+| `@popover-bg` | `popoverBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:122]:
```suggestion
| `@popover-bg` | `colorBgContainer` | Global Token |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 7]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-color` | `popoverColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:123]:
```suggestion
| `@popover-color` | `colorText` | Global Token |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 8]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-min-width` | `minWidth` | - |
+| `@popover-min-height` | `minHeight` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:125]: 可以废弃

File: docs/react/migrate-less-variables.en-US.md [hunk 9]
```diff
@@ -113,7 +113,21 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-arrow-width` | `arrowWidth` | - |
+| `@popover-arrow-color` | `arrowColor` | - |
+| `@popover-arrow-outer-color` | `arrowOuterColor` | - |
+| `@popover-distance` | `distance` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:129]:
```suggestion
| `@popover-distance` | `marginXXS` | Global Token |
```
先用这个替代

File: docs/react/migrate-less-variables.en-US.md (hunk 1 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:126]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 2 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:121]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 3 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:127]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 4 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:128]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 6 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:122]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 7 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:123]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 9 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:129]: done

File: docs/react/migrate-less-variables.zh-CN.md [hunk 10]
```diff
@@ -203,7 +203,20 @@ title: Less 变量迁移 Design Token
...
-<!-- ### Popover 气泡卡片 -->
+### Popover 气泡卡片
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@popover-bg` | `colorText` | - |
+| `@popover-color` | `colorText` | GlobalToken |
+| `@popover-min-width` | `minWidth` | - |
+| `@popover-min-height` | - | eprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:214]:
```suggestion
| `@popover-min-height` | - | Deprecated for style change |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 11]
```diff
@@ -205,7 +205,20 @@ This document contains the correspondence between all the less variables related
...
-<!-- ### Popover -->
+### Popover>
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@popover-bg` | `colorBgContainer` | GlobalToken |
+| `@popover-color` | `colorText` | GlobalToken |
+| `@popover-min-width` | `minWidth` | - |
+| `@popover-min-height` | - | eprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:216]:
```suggestion
| `@popover-min-height` | - | Deprecated for style change |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 12]
```diff
@@ -205,7 +205,20 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-arrow-width` | `sizePopupArrow` | GlobalToken |
+| `@popover-arrow-color` | - | eprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:218]:
```suggestion
| `@popover-arrow-color` | - | Deprecated for style change |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 13]
```diff
@@ -205,7 +205,20 @@ This document contains the correspondence between all the less variables related
...
+| `@popover-arrow-outer-color` | `arrowOuterColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:219]: 这个废弃了哈

File: docs/react/migrate-less-variables.zh-CN.md [hunk 14]
```diff
@@ -203,7 +203,20 @@ title: Less 变量迁移 Design Token
...
+| `@popover-arrow-width` | `sizePopupArrow` | 全局 Token |
+| `@popover-arrow-color` | - | eprecated for style change |
+| `@popover-arrow-outer-color`| - | eprecated for style change |
+| `@popover-distance` | `marginXXS` | Global Token |
+| `@popover-padding-horizontal` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:219]: 中文替换一下

File: components/popover/style/index.tsx [hunk 15]
```diff
@@ -5,21 +5,21 @@ import type { FullToken, GenerateStyle, PresetColorType } from '../../theme/inte
...
-  zIndexPopup: number;
-  width: number;
+  minWidth: number;
+  minHeight: number;
+  distance: number;
 }
 
 export type PopoverToken = FullToken<'Popover'> & {
-  popoverBg: string;
-  popoverColor: string;
```
Comment (MadCcc) [inline components/popover/style/index.tsx:14]: 这两个还是留着

File: components/popover/style/index.tsx [hunk 16]
```diff
@@ -5,21 +5,21 @@ import type { FullToken, GenerateStyle, PresetColorType } from '../../theme/inte
...
+  zIndexPopup: number;
```
Comment (MadCcc) [inline components/popover/style/index.tsx:14]: 这个还应该是组件 Token

File: components/popover/style/index.tsx [hunk 17]
```diff
@@ -5,21 +5,21 @@ import type { FullToken, GenerateStyle, PresetColorType } from '../../theme/inte
 import { PresetColors, genComponentStyleHook, mergeToken } from '../../theme/internal';
 
 export interface ComponentToken {
-  zIndexPopup: number;
-  width: number;
+  minWidth: number;
+  minHeight: number;
+  distance: number;
```
Comment (MadCcc) [inline components/popover/style/index.tsx:10]: 用 `marginXXS` 替代了，先不加

File: components/popover/style/index.tsx [hunk 18]
```diff
@@ -5,21 +5,21 @@ import type { FullToken, GenerateStyle, PresetColorType } from '../../theme/inte
 import { PresetColors, genComponentStyleHook, mergeToken } from '../../theme/internal';
 
 export interface ComponentToken {
-  zIndexPopup: number;
-  width: number;
+  minWidth: number;
+  minHeight: number;
```
Comment (MadCcc) [inline components/popover/style/index.tsx:9]: 这个也有迁移的话需要在文档里说明

Comment (MadCcc) [reply components/popover/style/index.tsx:9]: 而且看起来没有用过

File: components/popover/style/index.tsx [hunk 19]
```diff
@@ -5,21 +5,21 @@ import type { FullToken, GenerateStyle, PresetColorType } from '../../theme/inte
 import { PresetColors, genComponentStyleHook, mergeToken } from '../../theme/internal';
 
 export interface ComponentToken {
-  zIndexPopup: number;
-  width: number;
+  minWidth: number;
```
Comment (MadCcc) [inline components/popover/style/index.tsx:8]: 废弃原 token 参考 https://github.com/ant-design/ant-design/pull/42134/files#diff-e9607dd76735d305ab98d05211d7ebaf400d805eb8c30e2e667b7b5f27814811R383-R386

File: components/popover/style/index.tsx [hunk 20]
```diff
@@ -64,22 +66,22 @@ const genBaseStyle: GenerateStyle<PopoverToken> = (token) => {
...
-          minWidth: width,
+          minWidth,
           marginBottom: marginXS,
           color: colorTextHeading,
           fontWeight: fontWeightStrong,
         },
 
         [`${componentCls}-inner-content`]: {
-          color: popoverColor,
+          color: colorText,
```
Comment (MadCcc) [inline components/popover/style/index.tsx:84]:
```suggestion
          color: popoverColor,
```

File: components/popover/style/index.tsx [hunk 21]
```diff
@@ -64,22 +66,22 @@ const genBaseStyle: GenerateStyle<PopoverToken> = (token) => {
         },
 
         [`${componentCls}-inner`]: {
-          backgroundColor: popoverBg,
+          backgroundColor: colorBgContainer,
```
Comment (MadCcc) [inline components/popover/style/index.tsx:69]:
```suggestion
          backgroundColor: popoverBg,
```

File: components/popover/style/index.tsx [hunk 22]
```diff
@@ -181,8 +182,13 @@ export default genComponentStyleHook(
       initZoomMotion(popoverToken, 'zoom-big'),
     ];
   },
-  ({ zIndexPopupBase }) => ({
-    zIndexPopup: zIndexPopupBase + 30,
+  (token) => ({
     width: 177,
+    minWidth: 177,
+    minHeight: 32,
```
Comment (MadCcc) [inline components/popover/style/index.tsx:188]: 没有用过的话就不用加了

Comment (heiyu4585) [reply components/popover/style/index.tsx:188]: done

File: components/popover/style/index.tsx (hunk 21 above)
Comment (heiyu4585) [reply components/popover/style/index.tsx:69]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**